packages/webapp/src/containers/MediaWithAuthentication/index.jsx

```diff
@@ -18,30 +21,62 @@ export function MediaWithAuthentication({
       method: 'GET',
     };
     let subscribed;
-    const fetchMediaUrl = async () => {
+
+    const fetchMediaUrls = async () => {
       try {
         subscribed = true;
-        if (import.meta.env.VITE_ENV === 'development') {
-          subscribed && setMediaUrl(fileUrl);
+        if (mediaType === mediaEnum.ZIP) {
+          const zip = new JSZip();
+          const folder = zip.folder(title);
+
+          await Promise.all(
+            fileUrls.map(async (fileUrl) => {
+              const url = new URL(fileUrl);
+              if (import.meta.env.VITE_ENV !== 'development') {
+                url.hostname = 'images.litefarm.workers.dev';
+              }
+              return fetch(url.toString(), config).then((response) => {
+                const blobFilePromise = response.blob();
+                folder.file(url.href.substring(url.href.lastIndexOf('/')), blobFilePromise);
+              });
+            }),
+          );
+          const content = await zip.generateAsync({ type: 'base64' });
+          setZipContent(content);
         } else {
-          const url = new URL(fileUrl);
-          url.hostname = 'images.litefarm.workers.dev';
-          const response = await fetch(url.toString(), config);
-          const blobFile = await response.blob();
-          subscribed && setMediaUrl(URL.createObjectURL(blobFile));
+          const fileUrl = fileUrls[0];
```

Comment:
Kept the logic in this "else" block basically the same that it was before for single file downloads.